### PR TITLE
feat(explorer): abi fetch loaders

### PR DESCRIPTION
**Pull request by karooolis**
While fetching ABI, initially the ABI that's returned is not complete i.e. only includes system functions. Now showing a loader in navbar to indicate if the world is still being registerer. In addition, showing skeletons while "Interact" page is loading for better UX.

CleanShot 2024-08-28 at 13 57 58@2x

### CODE REVIEW

File: packages/explorer/src/components/Navigation.tsx [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 import { cn } from "../lib/utils";
+import { useAbiQuery } from "../queries/useAbiQuery";
 import { AccountSelect } from "./AccountSelect";
 
 export function Navigation() {
   const pathname = usePathname();
   const getLinkUrl = useWorldUrl();
+  const { data, isFetched } = useAbiQuery();
+
+  console.log(isFetched, data?.isWorldRegistered);
```
Comment (holic) [inline packages/explorer/src/components/Navigation.tsx:18]: delete?
```suggestion
```

Comment (karooolis) [reply packages/explorer/src/components/Navigation.tsx:18]: 😬

File: packages/explorer/src/app/api/world/route.ts [hunk 2]
```diff
@@ -27,9 +27,11 @@ async function getParameters(worldAddress: Address) {
     fromBlock: "earliest",
     toBlock,
   });
-  const fromBlock = logs[0].blockNumber;
+  const fromBlock = logs[0]?.blockNumber ?? 0n;
+  // world is considered loaded when both events are emitted
+  const isWorldRegistered = logs.length === 2;
 
-  return { fromBlock, toBlock };
+  return { fromBlock, toBlock, isWorldRegistered };
```
Comment (holic) [inline packages/explorer/src/app/api/world/route.ts:34]: `isWorldDeployed` might make more sense

File: packages/explorer/src/components/Navigation.tsx [hunk 3]
```diff
@@ -35,6 +39,12 @@ export function Navigation() {
           </Link>
         </div>
 
+        {isFetched && !data?.isWorldRegistered && (
+          <h4 className="font-mono text-sm font-bold uppercase opacity-70">
+            Registering world <Loader className="inline-block h-4 w-4 animate-spin" />
```
Comment (holic) [inline packages/explorer/src/components/Navigation.tsx:44]:
```suggestion
            Waiting for world deploy <Loader className="inline-block h-4 w-4 animate-spin" />
```

**Review by holic — APPROVED** [review]

**Pull request merged**